### PR TITLE
diorama camera rotation issue

### PR DESCRIPTION
**Pull request by tcm390**
related:
https://github.com/webaverse/app/issues/3423
https://github.com/webaverse/app/issues/3476

https://user-images.githubusercontent.com/60634884/182998716-03a25219-ae7d-4759-bdc7-bd647d6fe3a6.mp4

### CODE REVIEW

**Comment by tcm390** [comment]
[add lerp](https://github.com/webaverse/app/pull/3499/commits/ecbc2f5b3da47e355471b061d4a357420d365482)

https://user-images.githubusercontent.com/60634884/183101617-243af7cd-dbc4-4daf-8696-f9cd4bd5d4b4.mp4

**Review by lalalune — APPROVED** [review]

File: diorama.js [hunk 1]
```diff
@@ -702,6 +708,38 @@ const createPlayerDiorama = ({
...
+              
+              if(this.lastCrouch !== isCrouch || this.lastNR !== isNarutoRun){
+                this.lerpAlpha = 0.01;
+              }
+              else{
+                if(this.lerpAlpha < 1){
+                  this.lerpAlpha += 0.05;
+                }
+              }
+              sideCamera.position.x = lerp(sideCamera.position.x, sideCamera.position.x - (this.player.position.x - headPosition.x), this.lerpAlpha);
```
Comment (avaer) [inline diorama.js:732]: This could be `.set`?

File: diorama.js [hunk 2]
```diff
@@ -702,6 +708,38 @@ const createPlayerDiorama = ({
...
+          // handle player diorama
+          const lerp = (a, b, x) => {
+              return a + (b - a) * x;
+          }
+          if(this.player){
+            const headPosition = localVector3.setFromMatrixPosition(this.player.avatar.modelBones.Head.matrixWorld);
+            if(this.player.hasAction('sit')){
+              sideCamera.position.y -= this.player.position.y - headPosition.y;
+            }
+            else{
```
Comment (avaer) [inline diorama.js:720]: Spacing issues.

File: diorama.js [hunk 3]
```diff
@@ -702,6 +708,38 @@ const createPlayerDiorama = ({
             localVector2.set(0, cameraOffset.y, 0)
               .applyQuaternion(targetQuaternion)
           );
+          // handle player diorama
+          const lerp = (a, b, x) => {
+              return a + (b - a) * x;
+          }
+          if(this.player){
```
Comment (avaer) [inline diorama.js:715]: Is player required to make this look good? If so then it should be required 
 (can be set at construction time) and there is no need for this check?

Also, if there is a `.player` it should be initialized in the constructor regardless, even if it's just to ` null` (for performance reasons).

**Review by avaer — COMMENTED** [review]
Looks good other than some code quality issues.

**Comment by tcm390** [comment]
[clean code](https://github.com/webaverse/app/pull/3499/commits/431d191c9169450d41484f62e5b25dd5e452e82c)

File: diorama.js (hunk 1 above)
Comment (tcm390) [reply diorama.js]: https://github.com/webaverse/app/blob/32adf742456f9e16d8eab99ce488823362f5c834/diorama.js#L738

Comment (tcm390) [reply diorama.js]: change it to vector3.lerp 

https://github.com/webaverse/app/blob/32adf742456f9e16d8eab99ce488823362f5c834/diorama.js#L740

File: diorama.js (hunk 2 above)
Comment (tcm390) [reply diorama.js]: https://github.com/webaverse/app/blob/32adf742456f9e16d8eab99ce488823362f5c834/diorama.js#L726

File: diorama.js (hunk 3 above)
Comment (tcm390) [reply diorama.js]: Add `setPlayer`

https://github.com/webaverse/app/blob/32adf742456f9e16d8eab99ce488823362f5c834/diorama.js#L530

**Pull request closed without merging**